src/main/java/io/cryostat/core/reports/InterruptibleReportGenerator.java

```diff
@@ -185,6 +185,57 @@ public Future<ReportResult> generateReportInterruptibly(
                 });
     }
 
+    public Future<Map<String, ScoreRule>> generateScoreMapInterruptibly(
```

Comment:
Is there any refactoring that can be done so that this method's body isn't mostly just a copy-paste of the `generateReportInterruptibly`? There's a lot of shared code between that and this.

Reply:
> 

Maybe, the keys can be the IRule.Id(), and the value can still be a ScoreResult (or some other name), but with those extra fields of topics, names, plus the score and the short description. That way, we don't use a full IRule which has a field and method we don't care about. What do you think?